File: PatchMatch.m
function [offsets,matchTable] = PatchMatch(patchTable,kmatch)

    searchHalfWin = 19;
    
    if(nargin < 3)
        [offsets,matchTable] = InitTableRand(patchTable,searchHalfWin,kmatch);
    end

    numIt = 10;
    for i = 1:numIt
       disp(['iteration ' int2str(i)])
       % Propagate matches down and right on odd steps, up and left on even
        if(mod(i,2)==0)
            [offsets,matchTable] = PropagateEven(patchTable,offsets,matchTable);
        else
            [offsets,matchTable] = PropagateOdd(patchTable,offsets,matchTable);
        end
        mean(mean(matchTable(:,:,2)))
        [offsets,matchTable] = RandomSearch(patchTable,offsets,matchTable,searchHalfWin,1);
        mean(mean(matchTable(:,:,2)))
    end
    
end

File: Julia/patch_match.jl
require("propagate_even")
require("propagate_odd")
require("random_search")
require("init_table_rand")

function patch_match(patchTable,kmatch)

    searchHalfWin = 19
    (offsets,matchTable) = init_table_rand(patchTable,searchHalfWin,kmatch)
println("go1")
    numIt = 10
    for i = 1:numIt
       # Propagate matches down and right on odd steps, up and left on even
       if(iseven(i))
            println("go2")
           (offsets,matchTable) = propagate_even(patchTable,offsets,matchTable)
       else
            println("go3")
           (offsets,matchTable) = propagate_odd(patchTable,offsets,matchTable)
       end
       println(mean(matchTable[:,:,2]))
       (offsets,matchTable) = random_search(patchTable,offsets,matchTable,searchHalfWin,1);
       println(mean(matchTable[:,:,2]))
    end
    return offsets, matchTable
end
